plugins/inputs/docker/client.go

```diff
@@ -11,7 +11,7 @@ import (
 )
 
 var (
-	version        = "1.24" // https://docs.docker.com/engine/api/
+	version        = "1.43" // https://docs.docker.com/engine/api/
```

Comment:
The issue is that setting `DiskUsageOptions` is supported from versions 1.42 onwards.  See the 4th bullet point from the top at https://docs.docker.com/engine/api/version-history/#v142-api-changes. API version 1.42 corresponds to Docker Engine v23.0 (released in 02-2023). The options we have are:

1. Don't use `DiskUsageOptions` and always request everything. This might be problematic due to the API call being expensive for users with lots of containers, see: https://github.com/moby/moby/issues/31951
2. Set the API version within Telegraf to 1.42 and accept the consequences?
3. Instead of setting the version explicitly, we use https://pkg.go.dev/github.com/docker/docker/client#WithAPIVersionNegotiation and warn the user if the negotiated version is below 1.42?

What do you think?

Reply:
See commit https://github.com/influxdata/telegraf/pull/13894/commits/1db9c4f5849aff75206ebf59078def068d96f5e6 for an example implementation of variant 3 where the user is warned when the api version is unsupported.

Reply:
Sorry for the late reply. Yeah option 3 is the way to go IMO.